pkg/virtual/syncer/builder/template.go

```diff
@@ -208,20 +209,29 @@ func (t *template) bootstrapManagement(mainConfig genericapiserver.CompletedConf
 		t.cachedKCPInformers.Apis().V1alpha1().APIExports(),
 		func(syncTargetClusterName logicalcluster.Name, syncTargetName string, apiResourceSchema *apisv1alpha1.APIResourceSchema, version string, apiExportIdentityHash string) (apidefinition.APIDefinition, error) {
 			syncTargetKey := workloadv1alpha1.ToSyncTargetKey(syncTargetClusterName, syncTargetName)
-			requirements, selectable := labels.SelectorFromSet(map[string]string{
-				workloadv1alpha1.ClusterResourceStateLabelPrefix + syncTargetKey: string(t.filteredResourceState),
-			}).Requirements()
-			if !selectable {
-				return nil, fmt.Errorf("unable to build requirements for synctargetkey %s and resource state %s", syncTargetKey, t.filteredResourceState)
-			}
-			storageWrapper := t.storageWrapperBuilder(requirements)
-			transformingClient := t.dynamicClusterClient
-			if t.transformer != nil {
-				transformingClient = transforming.WithResourceTransformer(t.dynamicClusterClient, t.transformer)
-			}
 			ctx, cancelFn := context.WithCancel(context.Background())
-			storageBuilder := t.restProviderBuilder(ctx, transformingClient, apiExportIdentityHash, storageWrapper)
-			def, err := apiserver.CreateServingInfoFor(mainConfig, apiResourceSchema, version, storageBuilder)
+			var restProvider apiserver.RestProviderFunc
+			if apiResourceSchema.Spec.Group == kcpcorev1alpha1.SchemeGroupVersion.Group {
+				restProvider = NewSyncerRestProviderForLogicalClusters(ctx, t.dynamicClusterClient)
```

Comment:
will this be only those logical clusters that have scheduled workloads?

Reply:
No. Afaict it will return all the logical clusters of the shard.
I'll think about the best way to gather the workspaces that have scheduled workload on this SyncTarget.

Reply:
Removed since the logical cluster resources are not used nor exposed anymore